backend/src/test/java/capstone/backend/services/ProductServiceTest.java

```diff
@@ -163,6 +164,20 @@ void receiveGoodsFailsWhenOneOfItemsHasNegativeAmount() {
         assertThat(ex.getMessage(), is("You can't receive orders with negative quantity count"));
     }
 
+    @Test
+    void substractStockWhenAddingItemToBill() {
+        //GIVEN
+        OrderItem itemToAddToBill = sampleOrderItem();
+        Product productOnBill = itemToAddToBill.getProduct();
+        when(repo.getById(productOnBill.getId())).thenReturn(productOnBill.withAmountInStock(0));
+        when(repo.save(productOnBill)).thenReturn(productOnBill);
+        //WHEN
+        service.substractStockWhenAddingItemToBill(itemToAddToBill);
+        //THEN
+        verify(repo).getById(productOnBill.getId());
```

Comment:
How do you assert that the item is subtracted from stock?

Reply:
this method does not return anything, it just changes the value of the amountInStock prop of the respective product. Since this is a unit test and I mock everything that is not happening in the service, the only thing I can test for here is to verify that the methods in the repo that this method depens on have been called. In the integration tests I will also assert that the amountInStock of the product is actually lower than before